dynamic-checkbox-item/load_slack_channels_in_form_question.js

```diff
@@ -0,0 +1,80 @@
+var IGNORED_CHANNELS = [
+  'chat',
+  'slack-questions'
+]
+var FORM_ID = ''
+var SLACK_API_CHANNELS_URL = ''
+
+function onOpen () {
+  var form = FormApp.openById(FORM_ID)
+  var requestResponse = UrlFetchApp.fetch(SLACK_API_CHANNELS_URL)
+  var parsedJsonResponse = JSON.parse(requestResponse.getContentText())
+  var isSlackResponseOk = parsedJsonResponse.ok
+
+  if (!isSlackResponseOk) return
+
+  var eligibleChannels = prepareEligibleChannels(parsedJsonResponse)
+
+  var checkBoxItem = sanitizedCheckBoxItem(form)
+
+  eligibleChannels = eligibleChannels.map(function (channel) {
+    return checkBoxItem.createChoice(channel.name)
+  })
+
+  checkBoxItem
+    .setTitle('Select the channels you will probably join:')
+    .setChoices(eligibleChannels)
+}
+
+function prepareEligibleChannels (parsedJsonResponse) {
+  return parsedJsonResponse.channels.reduce(function (filteredChannels, channel) {
+    if (isChannelEligible(channel)) {
+      filteredChannels.push({
+        name: channel.name,
+        topic: channel.topic.value
+      })
+    }
+    return filteredChannels
+  }, [])
+}
+
+function isChannelEligible (channel) {
+  return !channel.is_general
+    && !channel.is_archived
+    && !channel.is_private
+    && channel.is_channel
+    && !isIgnoredChannel(channel)
+}
+
+function isIgnoredChannel (channel) {
+  return IGNORED_CHANNELS.indexOf(channel.name) > -1
+}
+
+function sanitizedCheckBoxItem (form) {
+  var checkBoxItem = findFirstItem(form)
```

Comment:
What if the first item is not a checkbox item? 

You should explain the the PR description how this works. To my understanding you are replacing the first item of the list with your dynamic checkbox item, right? Why not just add a new question to the form instead?